### DISCUSSION TITLE
CSV file schema and column level lineage is not captured

### DISCUSSION THREAD

**Discussion by Anitha-KS** [OpenLineage/OpenLineage#2568] [General]
from pyspark.sql import SparkSession

spark = (SparkSession.builder.master('local').appName('openlineage_spark_test')
    .getOrCreate())
spark.sparkContext.setLogLevel("TRACE")

empDF = spark.read.option("header",True).csv("file:///emp.1.txt") #.selectExpr("emp_id","name","superior_emp_id","year_joined","emp_dept_id","gender","salary")
empDF.printSchema()
empDF.count()

deptDF = spark.read.option("header",True).csv("file:///dept.1.txt") #.selectExpr("dept_name","dept_id")
deptDF.printSchema()

finalDF = empDF.join(deptDF,empDF.emp_dept_id ==  deptDF.dept_id,"inner")
finalDF.printSchema()
finalDF.write.options(header='True', delimiter=',').mode("overwrite").csv("ol_test_output")

This code is producing job level lineage, but
* the schema for input though captured not displaying in Marquez, shows only 1 column - value string.
* Column level lineage for the final output file seems not being generated based on logs and also not showing up in Marquez.

1. How to enable column lineage manually? 
2. For a controlled organization environment are there any spark settings that need to be done other than passing 4 elements in spark submit command?

**Comment by pawel-big-lebowski** [comment]
1. Currently there is no way for Spark integration to manually enrich events emitted. Feel free to create an issue for this. Perhaps more people will have similar requirements over time. 
2. Please refer to documentation [docs](https://openlineage.io/docs/integrations/spark/configuration/usage) on Spark configuration for more details. One has to add openlineage jar onto classpath, specify `io.openlineage.spark.agent.OpenLineageSparkListener` as extra listeners. Other required params are related to transport mechanism to emit OpenLineage events.

**Reply by Anitha-KS** [reply]
Thanks @pawel-big-lebowski !! 

I have followed the documentation and have added the required parameters, but still there is an issue. I will open an issue.

Before that, could you please share an example spark code(with column lineage)  and the expected Marquez output that can be used for POC. The example available today uses GCP so we are not able replicate the exact same one.

Thanks,
Anitha
